src/Annotations/AnnotationsRuleProvider.php

```diff
@@ -29,8 +29,6 @@ public function __construct(Reader $reader)
     }
 
     /**
-     * @param ReflectionClass|ReflectionMethod|ReflectionProperty $reflection
-     *
      * @return RuleInterface[]
      */
     public function getRules(Reflector $reflection): array
```

Comment:
This is one of the docblocks I had to remove to prevent analyse errors.

You have tests that make sure an exception is thrown if this method is called with a different type of reflector. Those tests create a mock of the `Reflector` interface, which PHPStan correctly complains about because it does not satisfy the `ReflectionClass|ReflectionMethod|ReflectionProperty` docblock annotation.

There may be some better way to approach these situations, but I'm not sure what it would be. Let me know if you have any ideas.

Reply:
I'd like to preserve this docblock for anyone who will read the code. I think I'll just add it to ignored errors in my phpstan config.